dvc/tree/s3.py

```diff
@@ -202,6 +202,7 @@ def _list_paths(self, path_info, max_items=None):
         kwargs = {"Prefix": path_info.path}
         if max_items is not None:
             kwargs["MaxKeys"] = max_items
+            kwargs["limit"] = max_items
```

Comment:
Thinking about it again, your alternative suggestion with `limit()` makes more sense. Even navigating the docs, this keyword is not really listed there hinting that this is not an official API, but `limit()` helper is. Let's use limit intead then.